### ISSUE TITLE
mixin-content

### ISSUE THREAD

**Issue by stephenway** [postcss/postcss-mixins#40]
Having an issue pretty consistently where I try to use `@mixin-content` and can never get it to replace with what I insert into the `@mixin` and it leaves the `@mixin-content` in the resulting CSS output.

I ran `gulp test` on a clone of this repo and that test comes up fine.

Here is my processors order:

``` js
        var processors = [
                cssimport,
                mixins,
                simplevars,
                nested,
                extend
        ];
```

**Comment by ai** [comment]
Show me please a CSS input and output

**Comment by stephenway** [comment]
Importing this https://github.com/stylep/stylep-card/blob/master/index.css
into this https://github.com/stylep/stylep-card/blob/master/example/index.css
which gives me this https://github.com/stylep/stylep-card/blob/master/example/build/index.css

**Comment by ai** [comment]
Can you show some small example? Input, output, expected output? (Sorry, I had too hard party after dotCSS yesterday)

**Comment by stephenway** [comment]
No problem, sounds like you had fun! :smile:

Input
```css
@define-mixin card-solid
  $color-passive: #f7f7f7,
  $color-text: #444,
  $radius-size: 0,
  $shadow: 0 2px 2px rgba(0, 0, 0, .4)
{
  background-color: $color-passive;
  border-radius: $radius-size;
  box-shadow: $shadow;
  color: $color-text;
  @mixin-content;
}

.card--solid {
  @mixin card-solid {
    color: green;
  }
}
```

Output
```css
.card--solid {
  background-color: #f7f7f7;
  border-radius: 0;
  box-shadow: 0 2px 2px rgba(0, 0, 0, .4);
  color: #444;
  @mixin-content
}
```

Expected output
```css
.card--solid {
  background-color: #f7f7f7;
  border-radius: 0;
  box-shadow: 0 2px 2px rgba(0, 0, 0, .4);
  color: #444;
  color: green;
}
```

**Comment by ai** [comment]
Fixed 2285b319d1001842e762fb5436d3eff7dfe021ca

**Comment by ai** [comment]
Released in 3.0.1

**Comment by stephenway** [comment]
Thanks!

**Comment by stephenway** [comment]
@ai Now it's not even compiling for me at `3.0.1` until comment out all `@mixin-content;` lines.

``` shell

events.js:141
      throw er; // Unhandled 'error' event
      ^
TypeError: Cannot read property 'type' of undefined
    at Root.normalize (/Users/Stephen/code/stylep-button/node_modules/postcss/lib/container.js:414:22)
    at Root.normalize (/Users/Stephen/code/stylep-button/node_modules/postcss/lib/root.js:41:52)
    at Root.insertBefore (/Users/Stephen/code/stylep-button/node_modules/postcss/lib/container.js:275:26)
    at AtRule.replaceWith (/Users/Stephen/code/stylep-button/node_modules/postcss/lib/node.js:155:29)
    at /Users/Stephen/code/stylep-button/node_modules/postcss-mixins/index.js:67:25
    at /Users/Stephen/code/stylep-button/node_modules/postcss/lib/container.js:153:34
    at /Users/Stephen/code/stylep-button/node_modules/postcss/lib/container.js:71:26
    at Root.each (/Users/Stephen/code/stylep-button/node_modules/postcss/lib/container.js:58:22)
    at Root.walk (/Users/Stephen/code/stylep-button/node_modules/postcss/lib/container.js:70:21)
    at Root.walkAtRules (/Users/Stephen/code/stylep-button/node_modules/postcss/lib/container.js:151:25)
```

**Comment by ai** [comment]
Can you remove copy your input CSS? I will fix it.

**Comment by stephenway** [comment]
Without flooding the thread I'll link to the file here. Thanks
https://raw.githubusercontent.com/stylep/stylep-button/master/index.css

**Comment by ai** [comment]
It complies on my mahcine. What plugins do you use? Maybe some plugin broke AST?

**Comment by stephenway** [comment]
Hmm, not sure, haven't looked into AST yet. https://github.com/stylep/stylep-button/blob/master/package.json#L10

**Comment by ai** [comment]
This source of issue was mixin without body.

Fixed: https://github.com/postcss/postcss-mixins/commit/29972d356dc35f3751fcefcb582da9b897e12974

**Comment by stephenway** [comment]
ahh, so if it wrap it in a `& {}` it will compile?

**Comment by ai** [comment]
Fix is released in 3.0.2. You can use old code, it was a postcss-mixin issue.

**Comment by stephenway** [comment]
Ok, I just tried that and it didn't work, but I'll upgrade instead. Thanks
